Modules/@babylonjs/react-native/windows/BabylonReactNative/EngineViewManager.cpp

```diff
@@ -0,0 +1,148 @@
+        _revokerData.PointerReleasedRevoker = _swapChainPanel.PointerReleased(winrt::auto_revoke, { this, &EngineViewManager::OnPointerReleased });
+
+        if (_revokerData.RenderingToken)
+        {
+            CompositionTarget::Rendering(_revokerData.RenderingToken);
+        }
+
+        _revokerData.RenderingToken = CompositionTarget::Rendering([weakThis{ this->get_weak() }](auto const&, auto const&)
+        {
+            if (auto trueThis = weakThis.get())
+            {
+                trueThis->OnRendering();
+            }
+        });
+
+        return _swapChainPanel;
+    }
+
+    // IViewManagerWithReactContext
+    IReactContext EngineViewManager::ReactContext() noexcept {
+        return _reactContext;
+    }
+
+    void EngineViewManager::ReactContext(IReactContext reactContext) noexcept {
+        _reactContext = reactContext;
+    }
+
+    // IViewManagerWithNativeProperties
+    IMapView<hstring, ViewManagerPropertyType> EngineViewManager::NativeProps() noexcept {
+        auto nativeProps = winrt::single_threaded_map<hstring, ViewManagerPropertyType>();
+
+        // TODO: Add properties as needed
+
+        return nativeProps.GetView();
+    }
+
+    void EngineViewManager::UpdateProperties(
+        FrameworkElement const& /*view*/,
+        IJSValueReader const& /*propertyMapReader*/) noexcept {
+
+        // TODO: Implement as needed
+    }
+
+    // IViewManagerWithExportedEventTypeConstants
+    ConstantProviderDelegate EngineViewManager::ExportedCustomBubblingEventTypeConstants() noexcept {
+        return nullptr;
+    }
+
+    ConstantProviderDelegate EngineViewManager::ExportedCustomDirectEventTypeConstants() noexcept {
+        return [](winrt::Microsoft::ReactNative::IJSValueWriter const& constantWriter) {
+            WriteCustomDirectEventTypeConstant(constantWriter, "onSnapshotDataReturned");
+        };
+    }
+
+    // IViewManagerWithCommands
+    IVectorView<hstring> EngineViewManager::Commands() noexcept {
+        auto commands = winrt::single_threaded_vector<hstring>();
+        commands.Append(L"takeSnapshot");
+        return commands.GetView();
+    }
+
+    void EngineViewManager::DispatchCommand(
+        FrameworkElement const& view,
+        winrt::hstring const& commandId,
+        IJSValueReader const& /*commandArgsReader*/) noexcept {
+        if (auto swapChainPanel = view.try_as<SwapChainPanel>()) {
+            if (commandId == L"takeSnapshot") {
+                // TODO: Implement takeSnapshot
+            }
+        }
+    }
+
+    void EngineViewManager::OnSizeChanged(IInspectable const& /*sender*/, SizeChangedEventArgs const& args)
+    {
+        const auto size = args.NewSize();
+        _swapChainPanelWidth = static_cast<size_t>(size.Width);
+        _swapChainPanelHeight = static_cast<size_t>(size.Height);
+
+        // Use windowTypePtr == 2 for xaml swap chain panels
+        auto windowTypePtr = reinterpret_cast<void*>(2);

```

Comment:
This is kind of unfortunate... We don't have a way to make this like an enum or something?

Reply:
We've talked about ways to make this type safe per platform in Babylon Native, but haven't fleshed out or implemented those ideas.

Reply:
FWIW, this code path doesn't seem well documented in bgfx. It already feels somewhat like a hack where folks overloaded a pointer needed for linux logic to do something random for windows logic.